common-practices-tools/security/yubikey.md

```diff
@@ -0,0 +1,309 @@
+
+```bash
+auth      required  pam_yubico.so   mode=challenge-response
+```
+
+##### YubiKey removal lock
+
+For additional security, you may want to immediately lock the screen when the YubiKey is removed.
+
+This locks the laptop immediately when any YubiKey is removed. If you are not using xautolock as your "away detector", replace xautolock with a command to trigger your screen lock with the "away detector" that you do use. This is inspired by <https://vtluug.org/wiki/Yubikey#Automatic_Screen_Locking_.28i3lock.2C_slock.2C_etc..29>
+
+As your login user, create executable file `~/bin/ykgone`:
+
+```bash
+#!/bin/bash
+USER=$(stat -c "%U" "$0")
+if usb-devices | fgrep Vendor=1050; then
+  echo "YubiKey present"
+else
+  echo "YubiKey not present, locking"
+  export DISPLAY=":0"
+  export XAUTHORITY=/home/$USER/.Xauthority
+  su $USER -c "xautolock -locknow" &
+fi
+```
+
+Next, create (with sudo) a device notification file `/etc/udev/rules.d/90-yubikey.rules`:
+
+```bash
+ACTION=="remove", ATTRS{idVendor}=="1050", RUN+="/home/$USER/bin/ykgone"
+```
+
+#### Personalize your YubiKey on Linux
+
+This allows you to use your YubiKey with Google TFA (new fangled U2F), as well as LastPass (which uses the OTP application).
+
+```bash
+neoman
+# Enable OTP, U2F, CCID checkboxes if needed, follow instructions to add and remove key
+
+ykpersonalize -2 -ochal-resp -ochal-hmac -ohmac-lt64 -oserial-api-visible
+```
+
+### Mac OS X
+
+This is the Mac OS X specific documentation for YubiKey.
+
+_Please help make this page more useful by adding links you found useful (describe exactly how they are useful) and specific steps you used to install, configure, and test your YubiKey._
+
+#### Personalize your YubiKey on MacOS
+
+Download and install the [YubiKey Personalization Tool](https://www.yubico.com/products/services-software/download/yubikey-personalization-tools/).
+
+This allows you to use your YubiKey with Google TFA (new fangled U2F), as well as LastPass (which uses the OTP application).
+
+#### Mac OSX YubiKey tool
+
+_This should be straightforward, but waiting for a pull request that clearly explains how to:_
+
+-   Enable `OTP`, `U2F` & `CCID`
+-   Personalize **Configuration Slot 2** with options:
+    -   `chal-resp` (Set challenge-response mode)
+    -   `chal-hmac` (Generate HMAC-SHA1 challenge responses)
+    -   `hmac-lt64` (Calculate HMAC on less than 64 bytes input)
+    -   `serial-api-visible` (Allow serial number to be read using an API call)
+
+#### Install apps
+
+See:
+
+-   <https://support.yubico.com/hc/en-us/articles/360013790219-Getting-Started-with-the-YubiKey-on-macOS>
+-   <https://www.yubico.com/why-yubico/for-businesses/computer-login/mac-os-login/>
+-   <https://www.yubico.com/2016/09/yubikey-smart-card-support-for-macos-sierra-2/>
+
+_tbd..._
+
+#### Set up PAM TFA on MacOS
+
+_tbd..._
+
+#### Screen lock on lid close or idle
+
+_tbd..._
+
+##### Away detection ideas
```

Comment:
This can become a two-octothorp (`##`) top-level section

Reply:
This section is specific to Mac or Linux users, so it's a sub-heading under that.